sql/catalyst/src/test/scala/org/apache/spark/sql/RandomDataGenerator.scala

```diff
@@ -172,7 +173,8 @@ object RandomDataGenerator {
               // January 1, 1970, 00:00:00 GMT for "9999-12-31 23:59:59.999999".
               milliseconds = rand.nextLong() % 253402329599999L
             }
-            DateTimeUtils.toJavaDate((milliseconds / MILLIS_PER_DAY).toInt)
+            val date = DateTimeUtils.toJavaDate((milliseconds / MILLIS_PER_DAY).toInt)
+            Try { date.toLocalDate; date }.getOrElse(new Date(date.getTime + MILLIS_PER_DAY))
```

Comment:
Shall we add a short comment that it adds one day in case the leap year is not matched?  Also, let's add some words that the dates should be both valid in hybrid Gregorian/Julian calendar and Proleptic Gregorian calendar.

Reply:
I added comments